xstats.go

```diff
@@ -69,17 +90,41 @@ func Copy(xs XStater) XStater {
 	return nop
 }
 
+// Scope makes a scoped copy of the given XStater if it implements the Scoper
+// interface. Otherwise it returns a nop stats.
+func Scope(xs XStater, scope string, scopes ...string) XStater {
+	if c, ok := xs.(Scoper); ok {
+		return c.Scope(scope, scopes...)
+	}
+	return nop
+}
+
 type xstats struct {
 	s Sender
 	// tags are appended to the tags provided to commands
 	tags []string
 	// prefix is prepended to all metric
 	prefix string
+	// delimiter is used to delimit scopes
+	delimiter string
 }
 
-// Copy makes a copy of the xstats client
+// Copy implements the Copier interface
 func (xs *xstats) Copy() XStater {
-	xs2 := NewPrefix(xs.s, xs.prefix).(*xstats)
+	xs2 := NewScoping(xs.s, xs.delimiter, xs.prefix).(*xstats)
+	xs2.tags = xs.tags
+	return xs2
+}
+
+// Scope implements Scoper interface
+func (xs *xstats) Scope(scope string, scopes ...string) XStater {
+	scs := make([]string, 0, 2+len(scopes))
```

Comment:
✅ also tried computing an int length first, and doing a single make. This is marginally more efficient.